src/client/app/containers/app.ts

```diff
@@ -198,6 +205,8 @@ export class AppComponent {
                     if (confirmationAlertIfAny !== this.strings.cancelButtonLabel) {
                         this._store.dispatch(new GitHub.SharePublicGistAction(this.snippet));
                     }
+
+                    this.isDisabled = false;
```

Comment:
Yes, it's necessary here, because in the event that the user presses the 'Cancel' option, the sharing state will never get changed because no action is dispatched. So 'isDisabled' needs to be set to still allow the menu to show.